build-tools/packages/build-cli/docs/typetestDetails.md

```diff
@@ -20,154 +22,73 @@ Type test generation is primarily configured in the package.json file of the pac
-| `baseMajor`            | 2.0.0-internal.2.0.0                          |
-| `baseMinor`            | 2.0.0-internal.2.3.0                          |
-| `~baseMinor`           | >=2.0.0-internal.2.3.0 <2.0.0-internal.3.0.0  |
-| `previousPatch`        | 2.0.0-internal.2.3.4                          |
-| `previousMajor`        | 2.0.0-internal.1.0.0                          |
-| `previousMinor`        | 2.0.0-internal.2.2.0                          |
-| `^previousMajor`       | >=2.0.0-internal.1.0.0 <2.0.0-internal.2.0.0  |
-| `^previousMinor`       | >=2.0.0-internal.2.2.0 <2.0.0-internal.3.0.0  |
-| `~previousMajor`       | >=2.0.0-internal.1.0.0 <2.0.0-internal.1.1.0  |
-| `~previousMinor`       | >=2.0.0-internal.2.2.0 <2.0.0-internal.2.3.0  |
-
-Given the version 2.0.0-internal.2.0.0:
-
-| Previous version style | Baseline version for **2.0.0-internal.2.0.0** |
-| ---------------------- | --------------------------------------------- |
-| `baseMajor`            | 2.0.0-internal.2.0.0                          |
-| `baseMinor`            | 2.0.0-internal.2.0.0                          |
-| `~baseMinor`           | >=2.0.0-internal.2.0.0 <2.0.0-internal.2.1.0  |
-| `previousPatch`        | 2.0.0-internal.2.0.0                          |
-| `previousMajor`        | 2.0.0-internal.1.0.0                          |
-| `previousMinor`        | 2.0.0-internal.2.0.0                          |
-| `^previousMajor`       | >=2.0.0-internal.1.0.0 <2.0.0-internal.2.0.0  |
-| `^previousMinor`       | >=2.0.0-internal.2.0.0 <2.0.0-internal.3.0.0  |
-| `~previousMajor`       | >=2.0.0-internal.1.0.0 <2.0.0-internal.1.1.0  |
-| `~previousMinor`       | >=2.0.0-internal.2.0.0 <2.0.0-internal.2.1.0  |
-
-## Pinning to a specific version
-
-This is useful when you want to regularly update the baseline version when releases happen, but only when explicitly
-updated, such as part of a release process. You can do this with a combination of a baseline range such as
-`^previousMajor` and the `--pin` flag.
-
-The `--pin` flag searches the release git tags in the repo and selects the baseline version as the maximum released
-version that matches the range. This effectively pins the version to a specific version while allowing it to be updated
-manually as needed by running type test preparation again.
-
-This is functionally similar to what a lockfile does, but this provides us with an extra level of control so we don't
-rely on lockfiles for type test package versions, because we have found bugs in lockfile handling related to npm
-aliases, which the type test infrastructure relies on.
+> [!NOTE]
+> The `generate:typetests` command is designed to produce a single output file - the type tests - per invocation, and to
+> generate tests for a single entrypoint. If you want to generate tests for both the alpha entrypoint and the legacy
+> entrypoint, for example, you will need to use `generate:typetests` twice, once for each entrypoint.
```

Comment:
I think this is the piece that has maybe confused people the most. I think we could make the setting an array and accept multiple values in the CLI flag in a future release, if we think that's more efficient or better than this design.

To note why I implemented it this way originally - one benefit to separate files is that it makes it easier to have different policies that apply to some type tests but not others. This is mostly relevant for CODEOWNERS, where we might want all changes to legacy type tests to be reviewed by a specific group.